solr/solrj/src/java/org/apache/solr/common/cloud/PerReplicaStatesOps.java

```diff
@@ -237,7 +237,7 @@ public static PerReplicaStatesOps downReplicas(List<String> replicas, PerReplica
         PerReplicaStates.State r = prs.get(replica);
         if (r != null) {
           if (r.state == Replica.State.DOWN && !r.isLeader) continue;
-          operations.add(new PerReplicaStates.Operation(PerReplicaStates.Operation.Type.ADD, new PerReplicaStates.State(replica, Replica.State.DOWN, Boolean.FALSE, r.version + 1)));
+          operations.add(new PerReplicaStates.Operation(PerReplicaStates.Operation.Type.ADD, new PerReplicaStates.State(replica, Replica.State.DOWN, r.isLeader, r.version + 1)));
```

Comment:
was it bug ? or is it caused by these changes?

Reply:
I think this is a minor bug that was generally hidden by other activity, I think a test failure caught this after I've removed other overseer -> PRS changes and there isn't a reason to remove leader when a replica is down since when it goes down it'll either:
1. Have leader election select a new leader if there is another possible core that can be leader
2. When it starts back up, already be leader in PRS state avoiding an issue in leader election on startup

It doesn't seem like we should change leader on node down, just the state, similar to what Overseer does in state.json: https://github.com/fullstorydev/lucene-solr/blob/d0577a2a4da12e10efdc42d2baef4afd592a6566/solr/core/src/java/org/apache/solr/cloud/overseer/NodeMutator.java#L78